tests/unit/test_http.py

```diff
@@ -102,31 +110,59 @@ async def test_session_stream(mock_request):
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_session_request_retry(mock_request, mock_response):
+async def test_session_request_retry(mock_request):
+    """Test the retry in the Session.request method"""
     async with http.Session() as ps:
         route = respx.get(TEST_URL)
         route.side_effect = [
             httpx.Response(HTTPStatus.TOO_MANY_REQUESTS, json={}),
             httpx.Response(HTTPStatus.OK, json={})
         ]
 
-        ps.retry_wait_time = 0  # lets not slow down tests for this
+        # let's not actually introduce a wait into the tests
+        ps.max_retry_backoff = 0
+
         resp = await ps.request(mock_request)
         assert resp
         assert route.call_count == 2
 
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_session_retry(mock_request):
-    async with http.Session() as ps:
+async def test_session__retry():
+    """A unit test for the _retry function"""
+
+    async def test_func():
+        # directly trigger the retry logic
+        raise exceptions.TooManyRequests
+
+    with patch('planet.http.Session._calculate_wait') as mock_wait:
```

Comment:
@sgillies here is where I have moved from mocking `asyncio.wait` to mocking `_calculate_wait` (hopefully a little less into the implementation weeds). `_calculate_wait` isn't async so we don't need to override the default with the `new` parameter in the patch fcn. However, I could still use help figuring out how to move patch from a context manager to a decorator.

Reply:
@sgillies I tried to follow the example you gave that worked for you, but still can't get a successful mock, whereas I'm successful with the context manager

Here's what I tried (the commented patch is one of my attempts too):
```python
@respx.mock
@pytest.mark.asyncio
# @patch("planet.http.Session._calculate_wait", autospec=True)
@patch.object(http.Session, '_calculate_wait', autospec=True)
async def test_session__retry(mock_wait):
    """A unit test for the _retry function"""
    # import pdb; pdb.set_trace()
    async def test_func():
        # directly trigger the retry logic
        raise exceptions.TooManyRequests

    # let's not actually introduce a wait into the tests
    mock_wait.return_value = 0

    async with http.Session() as ps:
        with pytest.raises(exceptions.TooManyRequests):
            await ps._retry(test_func)

    calls = mock_wait.call_args_list
    args = [c[0] for c in calls]
    assert args == [(1, 64), (2, 64), (3, 64), (4, 64), (5, 64)]
```

When I run this, I still see the fcn trying to wait for the wait times the original `_calculate_wait()` would return: 2, 4, 8, etc. instead of my mock, which returns 0.